llmfoundry/utils/builders.py

```diff
@@ -39,6 +40,30 @@
 log = logging.getLogger(__name__)
 
 
+def build_eval_loader(
+    eval_loader_config: DictConfig,
+    tokenizer: PreTrainedTokenizerBase,
+    device_eval_batch_size: int,
+) -> Evaluator:
+    evaluators = []
+
+    is_multi_eval = isinstance(eval_loader_config, ListConfig)
+    eval_configs = eval_loader_config if is_multi_eval else [eval_loader_config]
+    for eval_config in eval_configs:
+        eval_dataloader = build_dataloader(eval_config, tokenizer,
+                                           device_eval_batch_size)
+
+        # For training, metrics are added after the model is created
```

Comment:
commented elsewhere also, but I don't think this is quite right. we want to add the _train_ metrics for this "normal" evaluator, but the default of model.get_metrics() will give you _eval_ metrics if I'm reading the code correctly.